### PR TITLE
Tasks: Support new gas limit event handlers

### PR DESCRIPTION
**Pull request by PatricioHenderson**

### CODE REVIEW

**Comment by PatricioHenderson** [comment]
I didn't understood if we need to star tracking this 
`emit GasLimitsSet(newGasPriceLimit, newPriorityFeeLimit, newTxCostLimit, newTxCostLimitPct)`

**Comment by facuspagnuolo** [comment]
> I didn't understood if we need to star tracking this `emit GasLimitsSet(newGasPriceLimit, newPriorityFeeLimit, newTxCostLimit, newTxCostLimitPct)`

Yes, that's what we need

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "license": "GPL-3.0",
   "scripts": {
     "lint": "eslint . --ext .ts",
+    "codegen": "graph codegen",
```
Comment (facuspagnuolo) [inline package.json:8]: Was this intended?

```suggestion
```

File: schema.graphql [hunk 2]
```diff
@@ -109,10 +109,7 @@ type TaskConfig @entity{
   task: Task!
   nextBalanceConnector: String!
   previousBalanceConnector: String!
-  gasPriceLimit: BigInt!
-  priorityFeeLimit: BigInt!
-  txCostLimitPct: BigInt!
-  txCostLimit: BigInt!
+  gasLimits: [GasLimits!] @derivedFrom(field: "taskConfig")
```
Comment (facuspagnuolo) [inline schema.graphql:112]: Should be one, not a list

**Review by facuspagnuolo — COMMENTED** [review]
Looking good!

File: schema.graphql [hunk 3]
```diff
@@ -109,10 +109,7 @@ type TaskConfig @entity{
   task: Task!
   nextBalanceConnector: String!
   previousBalanceConnector: String!
-  gasPriceLimit: BigInt!
-  priorityFeeLimit: BigInt!
-  txCostLimitPct: BigInt!
-  txCostLimit: BigInt!
+  gasLimits: GasLimits!
```
Comment (facuspagnuolo) [inline schema.graphql:112]: Should be nullable

```suggestion
  gasLimits: GasLimits
```

File: src/Task.ts [hunk 4]
```diff
@@ -319,6 +331,20 @@ export function loadOrCreateAcceptanceList(tokensAcceptanceListId: string): Acce
...
 
+export function loadOrCreateGasLimits(gasPriceLimitsId: string): GasLimits {
+  let gasLimits = GasLimits.load(gasPriceLimitsId)
+
+  if (gasLimits === null) {
+    gasLimits = new GasLimits(gasPriceLimitsId)
+    gasLimits.gasPriceLimit = BigInt.zero()
+    gasLimits.priorityFeeLimit = BigInt.zero()
+    gasLimits.txCostLimitPct = BigInt.zero()
+    gasLimits.txCostLimit = BigInt.zero()
```
Comment (facuspagnuolo) [inline src/Task.ts:342]: Missing save

```suggestion
    gasLimits.txCostLimit = BigInt.zero()
    gasLimits.save()
```

**Review by facuspagnuolo — APPROVED** [review]

**Pull request merged**